src/dodal/devices/zebra.py

```diff
@@ -110,10 +110,9 @@ async def _set_armed(self, demand: ArmDemand):
             if reading == demand.value:
                 return
 
-    def set(self, demand: ArmDemand) -> AsyncStatus:
-        return AsyncStatus(
-            asyncio.wait_for(self._set_armed(demand), timeout=self.TIMEOUT)
-        )
+    @AsyncStatus.wrap
```

Comment:
lgtm, thanks.

I think we could move the code in `_set_armed` in this `set` here instead of keeping in split into two functions but I don't feel that strongly about it

Reply:
Yes, but I don't mind either way